README.md

```diff
@@ -57,48 +60,344 @@ This is work in progress as of 14 Nov 2018:
+| `successStatusCode`  | `200`              | HTTP status code for a successful response               |
+| `tags`               |                    | List of tags used for endpoint grouping in documentation |
+
+### `@request`
+
+Define a request body for requests that require one. This is commonly used for `POST` and `PUT` requests and is not allowed for `GET` requests:
+
+```TypeScript
+class MyUserEndpoints {
+  //...
+  @endpoint({
+    method: "POST",
+    path: "/users"
+  })
+  createUser(
+    @request req: CreateUserRequest
+  ): UserResponse {
+    return response();
+  }
+
+  @endpoint({
+    method: "PUT",
+    path: "/users/:id"
+  })
+  updateUser(
+    @pathParam({ description: "User unique identifier" }) id: string,
+    @request req: UpdateUserRequest
+  ): UserResponse {
+    return response();
+  }
+  //...
+}
+
+interface CreateUserRequest {
+  firstName: string;
+  lastName: string;
+}
+
+interface UpdateUserRequest {
+  lastName: string;
+}
+```
+
+### `@header`
+
+Define a request header. `@header` can be used multiple times to define multiple headers:
+
+```TypeScript
+  //...
+  @endpoint({
+    method: "POST",
+    path: "/users",
+    description: "Create a user"
+  })
+  createUser(
+    @request req: CreateUserRequest,
+    @header({
+      name: "Authorization",
+      description: "This is the authorization token"
+    })
+    authToken: Optional<string>
+  ): CreateUserResponse {
+    return response();
+  }
+  //...
+```
+
+| Field         | Description                       |
+| ------------- | --------------------------------- |
+| `name`        | (**required**) Name of the header |
+| `description` | Description of the header         |
+
+### `@pathParam`
+
+Define path parameters that appear in the `path` provided in `@endpoint()`. For example if the path is `/users/:userId`, the endpoint method must define a matching argument with `@pathParam() userId: string`:
+
+```TypeScript
+  //...
+  @endpoint({
+    method: "GET",
+    path: "/users/:id",
+    description: "Get user by id"
+  })
+  getUser(@pathParam({ description: "Unique user identifier" }) id: string): UserResponse {
+    return response();
+  }
+  //...
+```
+
+**Note**: the name of the argument must match the name of the path parameter.
+
+| Field         | Description                       |
+| ------------- | --------------------------------- |
+| `description` | Description of the path parameter |
+
+### `@queryParam`
+
+Define query parameters. `@queryParam` can be used multiple times to define multiple query parameters:
+
+```TypeScript
+  //...
+  @endpoint({
+    method: "GET",
+    path: "/users",
+    description: "Retrieve all users"
+  })
+  getUsers(@queryParam({ description: "Search term" }) search_term: Optional<string>): UserResponse[] {
+    return response();
+  }
+  //...
+```
+
+**Note**: the name of the argument must match the name of the query parameter.
```

Comment:
Is this true? I don't think there is any argument that needs to match the name of the query param. Maybe it is for path param